core/trino-main/src/main/java/io/trino/sql/iranalyzer/Scope.java

```diff
@@ -0,0 +1,428 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.sql.iranalyzer;
+
+import com.google.common.collect.ImmutableMap;
+import io.trino.spi.type.RowType;
+import io.trino.sql.ir.AllColumns;
+import io.trino.sql.ir.Expression;
+import io.trino.sql.ir.QualifiedName;
+import io.trino.sql.ir.WithQuery;
+
+import javax.annotation.concurrent.Immutable;
+
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.function.Predicate;
+
+import static com.google.common.base.MoreObjects.toStringHelper;
+import static com.google.common.base.Preconditions.checkArgument;
+import static com.google.common.collect.Iterables.getOnlyElement;
+import static io.trino.spi.StandardErrorCode.AMBIGUOUS_NAME;
+import static io.trino.sql.iranalyzer.ExpressionTreeUtils.asQualifiedName;
+import static io.trino.sql.iranalyzer.Scope.BasisType.FIELD;
+import static io.trino.sql.iranalyzer.Scope.BasisType.TABLE;
+import static io.trino.sql.iranalyzer.SemanticExceptions.ambiguousAttributeException;
+import static io.trino.sql.iranalyzer.SemanticExceptions.missingAttributeException;
+import static io.trino.sql.iranalyzer.SemanticExceptions.semanticException;
+import static java.util.Objects.requireNonNull;
+
+@Immutable
+public class Scope
```

Comment:
All the scope-related logic can be removed. It's only used during query analysis.

Reply:
done